url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/DATA/Filler.java

```diff
@@ -0,0 +1,980 @@
+package DATA;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import DATA.HashMap;
+import GUI.EdTViewerWindow;
+
+public class Filler	{
+
+	private ArrayList<Classroom> classrooms;
+	private ArrayList<ClassType> types;
+	private ArrayList<Group> groups;
+	private ArrayList<Teacher> teachers;
+	
+	private ArrayList<Lesson> done;						//Utilisée pour stocker l'"historique" pendant le remplissage.
+	private HashMap<Integer, Integer> steps;			//idem
+	private HashMap<Integer, Integer> errorsIndexes;	//idem
+	private int lastErrorIndex = 0;						//idem
+	
+	private Time MWWH; //Max Worked Week Hours, durée en heures de la semaine
+	private PrintStream ps;
+	
+	private int mode;
+	public static final int IGNORE = 0, ABORT = 1, RETRY = 2;
+	
+	public Filler(PrintStream aPs, ArrayList<Classroom> aClassrooms, ArrayList<Group> aGroups, ArrayList<Teacher> aTeachers, ArrayList<ClassType> aTypes, Time aMWWH/*, ArrayList<Constraint> aConstraints*/)	{
+	
+		this.classrooms = aClassrooms;
+		this.groups = aGroups;
+		this.teachers = aTeachers;
+		this.types = aTypes;
+		this.MWWH = aMWWH;
+		this.ps = aPs == null ? ps : aPs;
+	}
+	
+	public Filler(PrintStream aPs, DataStore ds) {
+		this.classrooms = ds.getClassrooms();
+		this.groups = ds.getGroups();
+		this.teachers = ds.getTeachers();
+		this.types = ds.getTypes();
+		this.MWWH = ds.getMWWH();
+		this.ps = aPs == null ? System.out : aPs;
+		
+	}
+	
+	/**
+	 * Dans l'ordre spécifié par la paramètre,
+	 * cette méthode devra attribuer les Constrainables
+	 * jusqu'à ce que l'emploi du temps soit complet.
+	 * Le paramètre mode détermine le comportement de la méthode handleError en cas d'échec.
+	 * @param order l'ordre général dans lequel traiter les données (déterminé par la méthode computeConstraints()).
+	 * @param mode Le comportement à appliquer si l'une des méthodes takeCareOf([...]) renvoie false ( <=> rencontre une erreur)
+	 * @return le nombre d'erreurs rencontrées lors du remplissage.
+	 */
+	public int fill(HashMap<Constrainable, Double> order, int mode)	{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 65-72 into a separate method.